spec/serializers/search_and_compare/course_serializer_spec.rb

```diff
@@ -18,6 +18,41 @@
         { subjects: subjects }
       end
 
+      let(:site_statuses_and_sites) do
+        site_statuses_and_sites = [
+          {
+            site_status_traits: %i[findable full_time_vacancies],
+            site_status_attrs: { applications_accepted_from: '2018-10-09T00:00:00' },
+            site_attrs: {
+              location_name: 'Stratford-Upon-Avon & South Warwickshire',
+              code: 'S',
+              address1: 'CV37',
+              address2: '',
+              address3: '',
+              address4: '',
+              postcode: '',
+
+            },
+            site_save_validate: false
+          },
+          {
+            site_status_traits: %i[findable full_time_vacancies],
+            site_status_attrs: { applications_accepted_from: '' },
+            site_attrs: {
+              location_name: 'Nuneaton & Bedworth',
+              code: 'N',
+              address1: 'CV10',
+              address2: '',
+              address3: '',
+              address4: '',
+              postcode: ''
+            },
+            site_save_validate: false
+          },
+        ]
+        { site_statuses_and_sites: site_statuses_and_sites }
+      end
```

Comment:
Is this kind of meta-representation of our test data really worth it? tbh, I've been wanting to find time to remove this kind of thing and replace it with just plain factory generation, so unless it's really ugly to do if with the existing factory system, I'd rather we just did that.

Reply:
I think you mentioning a number of things here.
`meta-representation`, I think its a good thing, ie grab another json, elicit the bits, put it into a course model and related models.

`plain factory generation/factory system`, I don't feel its applicable for this task as the default approach, as the task deals with actual stored data

data stored previously valid but currently it is invalid

eventually `plain factory generation/factory system` should be the the way to go

Reply:
https://github.com/DFE-Digital/manage-courses-backend/pull/403

I think this shows the folly of `plain factory generation` I could of omitted 2 of the 3 values that drives the 6 values. and the tests would of still passed

Reply:
I'm going to take a crack at this tonight. I'm still a bit reluctant to put in this kind of test generation stuff.

Reply:
thanks @misaka will continue to send my rubbish code to you